src/vaults/vaults.service.ts

```diff
@@ -170,16 +194,21 @@ export class VaultsService {
 
   public static getVaultYieldProjection(
     // refactor regV2, Data Objects interfaces should be in api alongside with models
-    vault: Pick<VaultDTO, 'value' | 'balance' | 'available' | 'lastHarvest'>,
+    vault: Pick<VaultDTO, 'value' | 'balance' | 'available' | 'lastHarvest' | 'underlyingToken'>,
     pendingHarvest: VaultPendingHarvestData,
   ): VaultYieldProjection {
     const { value, balance, available, lastHarvest } = vault;
+
     const harvestValue = pendingHarvest.harvestTokens.reduce((total, token) => (total += token.value), 0);
     const yieldValue = pendingHarvest.yieldTokens.reduce((total, token) => (total += token.value), 0);
+    const harvestCompoundValue = pendingHarvest.harvestTokens
```

Comment:
first change here - identify harvest components that are intended for compounding 

this gives us a better read on the true "compounding portion" for apy calculation